packages/cardano-services/src/PgBoss/stakePoolMetadataHandler.ts

```diff
@@ -46,8 +48,43 @@ export const savePoolMetadata = async (args: SavePoolMetadataArguments) => {
   }
 };
 
+export const getUrlToFetch = (
+  FETCH_MODE: StakePoolMetadataFetchMode,
+  SMASH_API_URL: string | undefined,
+  directUrl: string,
+  poolRegistrationId: string,
+  metadataHash: string
+) => {
+  if (FETCH_MODE === StakePoolMetadataFetchMode.SMASH) {
+    return `${SMASH_API_URL}/metadata/${poolRegistrationId}/${metadataHash}`;
+  } else if (FETCH_MODE === StakePoolMetadataFetchMode.DIRECT) {
+    return directUrl;
+  }
+  return directUrl;
+};
+
+export const attachExtendedMetadata = (
+  metadataWithoutExt: Cardano.StakePoolMetadata,
+  extMetadata: Cardano.ExtendedStakePoolMetadata | CustomError | undefined
+): Cardano.StakePoolMetadata => {
+  if (extMetadata instanceof CustomError) {
+    const error = extMetadata;
+
+    // This logic is here to stay complaint with the logic described in the interface StakePoolExtendedMetadataFields and used in DbSyncStakePoolProvider:attachExtendedMetadata.
+    if (error instanceof ProviderError && error.reason === ProviderFailure.ConnectionFailure) {
+      return { ...metadataWithoutExt!, ext: undefined };
```

Comment:
The two are not identical. Please take a look at the StakePoolExtendedMetadataFields interface.

Reply:
> The two are not identical

True, but:

```
const a = {};
const b = { prop: undefined };
```

differ for checks such as `'prop' in a` Vs `'prop' in b` or `Object.keys(a).length` Vs `Object.keys(b).length`, but not for checks such `if(a.prop)` Vs `if(b.prop)`, `a.prop === undefined` Vs `b.prop === undefined`.

Since with a return type `SomeType | null | undefined` the user anyway needs to perform following check
```
if(ret.prop) {} // handle `SomeType`
else if(ret.prop === null) {} // handle `null`
else {} // handle `undefined`
```
the SDK tolerates the approximation I'm suggesting.

The previous implementation of this handler itself was returning an object without the `ext` property rather with it set to `undefined`.

Reply:
Thanks @iccicci . Since you confirm that the previous implementation was handling this field differently than described in the StakePoolExtendedMetadataFields interface, I will gladly remove this eyesore.